src/core/validate-graph.ts

```diff
@@ -30,4 +32,45 @@ export function validateGraph(
     (pkgId) => !pkgNodes[pkgId] || pkgNodes[pkgId].size === 0,
   );
   assert(pkgsWithoutInstances.length === 0, 'not all pkgs have instance nodes');
+
+  for (const pkgId in pkgs) {
+    try {
+      validatePackageURL(pkgs[pkgId] as types.PkgInfo);
+    } catch (e) {
+      throw new ValidationError(`invalid pkg ${pkgId}: ${e}`);
+    }
+  }
+}
+
+export function validatePackageURL(pkg: types.PkgInfo): void {
+  if (!pkg.purl) {
+    return;
+  }
+
+  try {
+    const purlPkg = PackageURL.fromString(pkg.purl);
+
+    switch (purlPkg.type) {
+      // Within Snyk, maven packages use <namespace>:<name> as their *name*, but
+      // we expect those to be separated correctly in the PackageURL.
+      case 'maven':
```

Comment:
'gradle' and 'sbt'?

Reply:
No idea, @darscan? 

(Even if, this would only become an issue once the respective teams decide to add the purl, so it shouldn't break anything on release)

Reply:
Let's defer. The PURL "types" for Gradle and SBT have not been defined yet: https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#other-candidate-types-to-define